### ISSUE TITLE
Uncaught TypeError: Cannot read property 'firstCharIndentation' of undefined

### ISSUE THREAD

**Issue by joshpitzalis** [gandm/language-babel#384]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.18.0 x64
**Electron**: 1.3.15
**OS**: Mac OS X 10.12.5
**Thrown From**: [language-babel](https://github.com/gandm/language-babel) package 2.66.0

### Stack Trace

Uncaught TypeError: Cannot read property 'firstCharIndentation' of undefined

```
At /Users/josh/.atom/packages/language-babel/lib/auto-indent.coffee:433

TypeError: Cannot read property 'firstCharIndentation' of undefined
    at AutoIndent.module.exports.AutoIndent.indentJSX (/packages/language-babel/lib/auto-indent.coffee:433:91)
    at AutoIndent.module.exports.AutoIndent.didStopChanging (/packages/language-babel/lib/auto-indent.coffee:141:10)
    at /packages/language-babel/lib/auto-indent.coffee:151:66
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:1442:31
```

### Commands

```
  2x -3:35.5.0 core:paste (input.hidden-input)
     -3:29.6.0 core:move-right (input.hidden-input)
  5x -3:29.4.0 core:backspace (input.hidden-input)
     -3:27 core:undo (input.hidden-input)
     -3:24.9.0 core:paste (input.hidden-input)
 17x -2:38.8.0 core:backspace (input.hidden-input)
     -2:19.5.0 core:save (input.hidden-input)
     -2:19.4.0 linter:lint (atom-text-editor.editor.is-focused.hyperclick)
     -2:17.4.0 editor:consolidate-selections (input.hidden-input)
     -2:17.4.0 core:cancel (input.hidden-input)
  2x -0:40.8.0 tree-view:remove (span.name.icon.icon-file-directory)
     -0:17.9.0 emmet:toggle-comment (input.hidden-input)
     -0:17.9.0 editor:toggle-line-comments (input.hidden-input)
     -0:15.7.0 core:save (input.hidden-input)
     -0:07.0 core:copy (input.hidden-input)
  2x -0:05.9.0 editor:newline (input.hidden-input)
```

### Non-Core Packages

```
autocomplete-paths 1.0.5 
autoprefixer 3.7.0 
busy-signal 1.4.3 
emmet 2.4.3 
hyperclick 0.0.40 
intentions 1.1.2 
js-hyperclick 1.11.0 
language-babel 2.66.0 
linter 2.2.0 
linter-ui-default 1.6.2 
pigments 0.39.1 
prettier-atom 0.34.0 
```

**Comment by gandm** [comment]
If you can reproduce this can I have the JSX in question.

**Comment by lucasfeliciano** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.17.2 x64
**Electron**: 1.3.15
**OS**: Mac OS X 10.12.5
**Thrown From**: [language-babel](https://github.com/gandm/language-babel) package 2.66.0


### Stack Trace

Uncaught TypeError: Cannot read property 'firstCharIndentation' of undefined

```
At /Users/lucasfeliciano/.atom/packages/language-babel/lib/auto-indent.coffee:433

TypeError: Cannot read property 'firstCharIndentation' of undefined
    at AutoIndent.module.exports.AutoIndent.indentJSX (/packages/language-babel/lib/auto-indent.coffee:433:91)
    at AutoIndent.module.exports.AutoIndent.changedCursorPosition (/packages/language-babel/lib/auto-indent.coffee:113:6)
    at AutoIndent.changedCursorPosition (/packages/language-babel/lib/auto-indent.coffee:1:1)
    at /packages/language-babel/lib/auto-indent.coffee:73:68
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at TextEditor.module.exports.TextEditor.cursorMoved (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2136:1)
    at Selection.module.exports.Selection.markerDidChange (/Applications/Atom.app/Contents/Resources/app/src/selection.js:921:1)
    at /Applications/Atom.app/Contents/Resources/app/src/selection.js:41:1
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at DisplayMarker.module.exports.DisplayMarker.notifyObservers (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:248:1)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:53:1
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at Marker.module.exports.Marker.emitChangeEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:421:1)
    at Marker.module.exports.Marker.update (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:362:1)
    at Marker.module.exports.Marker.setHeadPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/marker.js:144:1)
    at DisplayMarker.module.exports.DisplayMarker.setHeadBufferPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:146:1)
    at DisplayMarker.module.exports.DisplayMarker.setHeadScreenPosition (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/display-marker.js:154:1)
    at /Applications/Atom.app/Contents/Resources/app/src/cursor.js:72:1
    at Cursor.module.exports.Cursor.changePosition (/Applications/Atom.app/Contents/Resources/app/src/cursor.js:684:1)
    at Cursor.module.exports.Cursor.setScreenPosition (/Applications/Atom.app/Contents/Resources/app/src/cursor.js:70:1)
    at /Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1907:1
    at TextEditor.module.exports.TextEditor.moveCursors (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2130:1)
    at TextEditor.module.exports.TextEditor.setCursorScreenPosition (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1906:1)
    at TextEditorComponent.module.exports.TextEditorComponent.onMouseDown (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:733:1)
    at HTMLDivElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:3:1)
```

### Commands

```
     -2:07.8.0 fuzzy-finder:toggle-file-finder (input.hidden-input)
     -2:05.5.0 core:confirm (input.hidden-input)
     -1:15 core:cut (input.hidden-input)
  2x -1:12.4.0 editor:newline (input.hidden-input)
  7x -1:10.8.0 core:backspace (input.hidden-input)
     -0:59.2.0 autocomplete-plus:confirm (input.hidden-input)
     -0:52.7.0 core:move-right (input.hidden-input)
     -0:50.9.0 core:backspace (input.hidden-input)
     -0:48.4.0 editor:newline (input.hidden-input)
  2x -0:47.7.0 core:paste (input.hidden-input)
 22x -0:41.8.0 core:undo (input.hidden-input)
     -0:10.2.0 editor:consolidate-selections (input.hidden-input)
     -0:10.2.0 core:cancel (input.hidden-input)
```

### Non-Core Packages

```
atom-material-syntax 1.0.6 
atom-material-ui 2.0.4 
behat-atom 0.1.2 
busy-signal 1.4.3 
duotone-dark-space-syntax 2.1.0 
duotone-dark-syntax 2.1.0 
duotone-darksky 1.1.5 
emmet 2.4.3 
git-plus 7.9.3 
haskell-grammar 0.4.0 
hey-pane 1.0.0 
highlight-selected 0.13.1 
intentions 1.1.2 
language-babel 2.66.0 
language-graphql 0.9.0 
language-ini 1.19.0 
language-javascript-jsx 0.3.7 
language-lua 0.9.11 
language-ocaml 1.9.5 
language-rust 0.4.10 
language-swift 0.5.0 
language-thrift 1.0.2 
linter 2.2.0 
linter-eslint 8.2.1 
linter-ui-default 1.6.2 
MagicPython 1.0.10 
merge-conflicts 1.4.5 
minimap 4.28.2 
minimap-highlight-selected 4.6.1 
nuclide-format-js 0.0.36 
php-twig 4.0.0 
prettier-atom 0.34.0 
react-es6-snippets 0.3.0 
react-snippets 0.8.0 
set-syntax 0.3.2 
sort-lines 0.14.0 
toggle-quotes 1.0.1 
tool-bar 1.1.0 
```

**Comment by gandm** [comment]
@lucasfeliciano I could put a check for a undefined property that would stop this problem however it shouldn't be undefined. I would really like the JSX and the steps that caused this so I can get to the root cause. It would be great if the was reproducible.

**Comment by joshpitzalis** [comment]
This seems to be happening in atom in any create-react-app project that I add react-router-dom to. I can't share the original because it's in someone else's private repo but I made another project yesterday and the same thing is happening again. https://github.com/joshpitzalis/hive/blob/master/frontend/src/components/routes.js

**Comment by gandm** [comment]
@joshpitzalis Thanks for that. I hope this is fixed in 2.66.1

**Comment by joshpitzalis** [comment]
Yes. It works perfectly now. You are amazing. Thank you very much.
